ncm-resolver: fails to report an error if the lookup times out

https://github.com/quattor/configuration-modules-core/blob/master/ncm-resolver/src/main/perl/resolver.pm#L115

If `$out =~ /timed out/` then no sensible error is produced, e.g. if `host` reports to stdout:

```
;; connection timed out; no servers could be reached
```

The best we get from `ncm-resolver` is:

```
[INFO] running component: resolver
---------------------------------------------------------
[ERROR] will not change resolv.conf; looking up <host> on <server> fails with:
[INFO] configure on component resolver executed, 1 errors, 0 warnings
```

--- a/ncm-resolver/src/main/perl/resolver.pm
+++ b/ncm-resolver/src/main/perl/resolver.pm
@@ -107,12 +107,11 @@ sub Configure {
     # we commit to this.
     $self->log("using $host to test our dns config");
     my $out = "";
-    my $errs = "";
     my $rc = LC::Process::execute(["/usr/bin/host", $host, $testserver],
-                                stderr => \$errs,
+                                stderr => 'stdout',
                                 stdout => \$out);
     if (!$rc || $out =~ /timed out/) {
-        $self->error("will not change resolv.conf; looking up $host on $testserver fails with: $errs");
+        $self->error("will not change resolv.conf; looking up $host on $testserver fails with output: $out");
         if ($inf->{dnscache}) {
             # We need to put the dnscache config back to the way it was
             $self->change_dnscache($inf, $servers_file, $old);
@@ -151,10 +150,10 @@ sub change_dnscache {
             $self->log("$servers_file unchanged");
         } else {
             $self->log("updated $servers_file");
-    
+
             my $errs = "";
             my $out= "";
-            my $rc = LC::Process::execute(["/etc/init.d/dnscache", "restart" ], stderr => \$errs, stdout => \$out); 
+            my $rc = LC::Process::execute(["/etc/init.d/dnscache", "restart" ], stderr => \$errs, stdout => \$out);
             $self->debug(5, "restart dnscache said: $out");
             if (!$rc) {
                 $self->error("failed to restart dnscache: $errs");

